### DISCUSSION TITLE
How does msfvenom use an existing .exe file as a template?

### DISCUSSION THREAD

**Discussion by Morsmalleo** [rapid7/metasploit-framework#16699] [Q&A]
Another quick question here guys, similar to my one about APK's but this one is for EXE's as both standalone Payloads and as templates.

Does anyone happen to know how msfvenom generates Windows `.exe` payloads? I'd also like to know how msfvenom is able to Bind an existing Windows executable (`.exe`) payload to an original Windows Executable (`.exe`) File, if anyone happens to know.

Any answers on this would be greatly appreciated!

**Comment by Morsmalleo** [comment]
> Another quick question here guys, similar to my one about APK's but this one is for EXE's as both standalone Payloads and as templates.
> 
> Does anyone happen to know how msfvenom generates Windows `.exe` payloads? I'd also like to know how msfvenom is able to Bind an existing Windows executable (`.exe`) payload to an original Windows Executable (`.exe`) File, if anyone happens to know.
> 
> Any answers on this would be greatly appreciated!

Still hoping for an answer on this one.

**Comment by timwr** [comment]
I think it depends on the arguments you give to msfvenom, but it looks like it tries to add a new section to the exe:
https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/util/exe.rb#L246
